nototools/tool_utils.py

```diff
@@ -314,45 +314,118 @@ def svn_update(repo):
 
 
 def parse_int_ranges(
-    range_string, is_hex=True, sep=None, allow_duplicates=False):
-  """Returns a set of ints from a string of numbers or ranges separated by sep.
-  A range is two values separated by hyphen with no intervening separator;
-  ranges are inclusive."""
-  result = set()
-  count = 0
+    range_string, is_hex=True, sep=None, allow_duplicates=False,
+    return_set=True, allow_compressed=False):
+  """Returns a set/list of ints from a string of numbers or ranges separated by
+  sep.  A range is two values separated by hyphen with no intervening separator;
+  ranges are inclusive.  If allow_compressed is true, '/' is also allowed
+  as a separator, and ranges following it or hyphen are interpreted as suffixes
+  that replace the same number of characters at the end of the previous value.
+  '-' generates the range of intervening characters as before, while '/' does
+  not.  Returns a set or a list depending on return_set."""
+
   base = 16 if is_hex else 10
+  vals = []
+
+  def _add_segment(prev_str, suffix, is_range):
```

Comment:
Just a thought (no need to change):  Isn't  parse_int_ranges() too long? Wouldn't it be better to move two nested functions out of it?